### DISCUSSION TITLE
Monitor which disk(service) is down

### DISCUSSION THREAD

**Discussion by hamidreza-hosseini** [seaweedfs/seaweedfs#2507] [Q&A]
Hi,
I've got a problem with weed monitoring,
Each disk is a weed-volume service in my cluster, every time that a server  or disk(service) goes down I couldn't understand this,
I should check "cluster status UI " every day and see max and free volumes to see if a server or disk is out of service

1. How can I set an alert to understand server or disk(service) outages and
2. Which server and service are not working ?

**Comment by chrislusf** [comment]
The master would not necessarily know which volume server is down, since it only passively receive status reported by the volume servers themselves.

The design was to let volume servers freely move around. So the master would not know whether a volume server is moved or dead.

**Reply by hamidreza-hosseini** [reply]
> The master would not necessarily know which volume server is down, since it only passively receive status reported by the volume servers themselves.
> 
> The design was to let volume servers freely move around. So the master would not know whether a volume server is moved or dead.

I know that but in production, we need status or an alert to find out which volume servers have gone or are out of service now ,
because there is a limited replica count for each volume e.g: 3 for each volume, what if 4 servers go down and I don't understand this?
objects couldn't serve due to outages
as far as I know, all object servers have this option, for example, Ceph sends its status to Prometheus, and then it will alert if anything bad happened ...
We necessarily need this option in weed
Many Thanks.

**Reply by chrislusf** [reply]
volume servers have metrics sent to prometheus. You can setup that and use it to track.

**Reply by hamidreza-hosseini** [reply]
Except this, there is no way for alerting and monitoring?

**Reply by chrislusf** [reply]
You can write a script to list the volume servers in "weed script", and alert if any one is missing.
